### PR TITLE
Add fit methods for tranquilo

### PR DESCRIPTION
**Pull request by timmens**
In this PR we implement basic fitting methods for the surrogate models used in TRANQUILO.

### CODE REVIEW

File: src/estimagic/optimization/tranquilo/surrogate_models.py [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+    centered_f,
+    fitter,
+    options,
+):
+    """Fit a model to the data.
+
+    Args:
+        fitter (str or callable): Name of a fit method or a fit method. The first
+            argument of any fit method needs to be ``x``, the second ``f``.
+        user_options (dict): Options for the fit method. The following are supported:
```
Comment (timmens) [inline src/estimagic/optimization/tranquilo/surrogate_models.py:100]: options not user_options

File: src/estimagic/optimization/tranquilo/surrogate_models.py [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+    return out
+
+
+def _fitter_template(
+    centered_x,
+    centered_f,
+    fitter,
+    options,
+):
+    """Fit a model to the data.
```
Comment (timmens) [inline src/estimagic/optimization/tranquilo/surrogate_models.py:95]: to data

File: tests/optimization/tranquilo/test_surrogate_models.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        "x0": x0,
+        "x": x,
+        "y": y,
+        "linear_terms_expected": linear_terms,
+        "square_terms_expected": square_terms,
+    }
+    return out
+
+
+def test_fit_ols_agfainst_truth(quadratic_case):
```
Comment (timmens) [inline tests/optimization/tranquilo/test_surrogate_models.py:60]: against_truth

**Review by janosg — APPROVED** [review]
Thanks!

**Pull request merged**